doc/source/index.rst

```diff
@@ -24,7 +24,7 @@ Sections
    license
    auto_examples/index
    release_notes_and_installation
-
+   core_developer
```

Comment:
>That repo needs to be fleshed out a bit before we commit to it. My suggestion would be to get this merged ASAP (focusing on the substance of the text rather than where it should go) and then later make an effort to reorganise.

I'm glad we have an agreement here :). Perhaps, in the future this should link to a complete list of the organizational documents.